File: 2º Ano/POO/ficha3/ex3/ex3/Retangulo.cpp
//
//  Retangulo.cpp
//  ex3
//
//  Created by rsantos on 03/11/2021.
//

#include "Retangulo.hpp"

Retangulo::Retangulo(Ponto cse, int comp, int alt) : cantoSE(cse), comprimento(comp), altura(alt) {
    
}

string Retangulo::getAsString() const {
    ostringstream os;
    os << "altura: " << altura << "; comprimento: " << comprimento << "\n" << endl;
    os << "(" << cantoSE.getX() << "," << cantoSE.getY() << ")\t\t(" << cantoSE.getX() + comprimento << "," << cantoSE.getY() << ")" << endl;
    os << "(" << cantoSE.getX()-altura << "," << cantoSE.getY() << ")\t\t(" << cantoSE.getX() << "," << cantoSE.getY() + comprimento << ")" <<endl;
    return os.str();
}

File: 3º Ano/ED/Pratica/Revisao/Teste1/ficha3/src/pt/isec/ed/ex3/Retangulo.java
package pt.isec.ed.ex3;

public class Retangulo extends Figura {
    private int comprimento, largura;

    public Retangulo(int comprimento, int largura) {
        this.comprimento = comprimento;
        this.largura = largura;
    }

    @Override
    float area() { return comprimento * largura; }
}
